domain/consensus/processes/consensusstatemanager/reverse_utxo_diffs.go

```diff
@@ -0,0 +1,92 @@
+package consensusstatemanager
+
+import (
+	"github.com/kaspanet/kaspad/domain/consensus/model"
+	"github.com/kaspanet/kaspad/domain/consensus/model/externalapi"
+	"github.com/kaspanet/kaspad/infrastructure/logger"
+	"github.com/kaspanet/kaspad/util/staging"
+)
+
+func (csm *consensusStateManager) ReverseUTXODiffs(tipHash *externalapi.DomainHash,
+	reversalData *model.UTXODiffReversalData) error {
+
+	// During the process of resolving a chain of blocks, we temporarily set all blocks' (except the tip)
+	// UTXODiffChild to be the selected parent.
+	// Once the process is complete, we can reverse said chain, to now go directly to virtual through the relevant tip
+
+	onEnd := logger.LogAndMeasureExecutionTime(log, "reverseUTXODiffs")
+	defer onEnd()
+
+	readStagingArea := model.NewStagingArea()
```

Comment:
Maybe rename it to emptyStagingArea?

Reply:
I prefer "readStagingArea" because it's only used for reads (and the writes happen in a separate staging area).